src/Collection.php

```diff
@@ -94,7 +96,13 @@ public static function range($start = 0, $end = null, $step = 1)
     public function getIterator()
     {
         if ($this->inputFactory) {
-            $this->input = call_user_func($this->inputFactory);
+            $input = call_user_func($this->inputFactory);
+
+            if (is_array($input)) {
+                $input = new RecursiveArrayIterator($input);
+            }
+
```

Comment:
Should we throw if `$input` is no `Traversable` at this point? 
It should have raised an exception in the constructor already, right?